airflow/kubernetes/kubernetes_request_factory/pod_request_factory.py

```diff
@@ -15,12 +15,47 @@
 # specific language governing permissions and limitations
 # under the License.
 
-from typing import Dict
 
 import yaml
-from airflow.kubernetes.pod import Pod
-from airflow.kubernetes.kubernetes_request_factory.kubernetes_request_factory \
-    import KubernetesRequestFactory
+from abc import ABCMeta
+from kubernetes.client import ApiClient
+import kubernetes.client.models as k8s
+from typing import Dict
+
+
+def merge_dicts(pod_dict: Dict, base_dict: Dict) -> Dict:
```

Comment:
Not quite - this does it "deeply" where as dict.update doesn't.

If we want to handle lists too then it may be worth bringing in https://github.com/toumorokoshi/deepmerge

Reply:
I see. I am surprised that the standard lib doesn't support this kind of "JSON merge patch"-like behavior natively, though. 😕